Sources/Confidence/EventStorage.swift

```diff
@@ -0,0 +1,104 @@
+import Foundation
+
+internal protocol EventStorage {
+    func rollover() throws
+    func writeEvent(event: Event) throws
+    func batchReadyFiles() throws -> [URL]
+    func eventsFrom(fileURL: URL) throws -> [Event]
+}
+
+internal class EventStorageImpl: EventStorage {
+    let DIRECTORY = "events"
+    let EVENT_WRITE_DELIMITER = ",\n"
+    let READY_TO_SEND_EXTENSION = ".ready"
+    let encoder = JSONEncoder()
+    let decoder = JSONDecoder()
+    var folderURL: URL = URL(string: "")!
+    var fileURL: URL = URL(string: "")!
+    var currentBatch: [Event] = []
+
+    init() throws {
+        folderURL = try getFolderURL()
+        try createEventsDirectory(url: folderURL)
+        fileURL = folderURL.appendingPathComponent("events-\(Date().currentTime)")
+    }
+
+    func rollover() throws {
+        let newPath = URL(string: "\(fileURL)" + "\(Date().currentTime)")!
+        try FileManager.default.moveItem(at: fileURL, to: newPath)
+        try FileManager.default.removeItem(at: fileURL)
+        fileURL = folderURL.appendingPathComponent("events-\(Date().currentTime)")
+        currentBatch = []
+    }
+    
+    func writeEvent(event: Event) throws {
+        currentBatch.append(event)
+        let data = try encoder.encode(currentBatch)
+        try data.write(to: fileURL, options: .withoutOverwriting)
+    }
+    
+    func batchReadyFiles() throws -> [URL] {
+        var readyFilesList: [URL] = []
+        let directoryContents = try FileManager.default.contentsOfDirectory(atPath: folderURL.absoluteString)
+        for file in directoryContents {
+            if file.hasSuffix(READY_TO_SEND_EXTENSION) {
```

Comment:
You never set this suffix on files so this will never be true.
I assume that this is what you actually want to append in `rollover()`?

Reply:
Indeed 👍